rslib/src/sync/http_server/mod.rs

```diff
@@ -91,9 +96,26 @@ impl SimpleServerInner {
             match std::env::var(&envvar) {
                 Ok(val) => {
                     let hkey = derive_hkey(&val);
-                    let (name, _) = val.split_once(':').with_whatever_context(|| {
-                        format!("{envvar} should be in 'username:password' format.")
-                    })?;
+                    let (name, pwhash) = {
+                        let (name, password) = val.split_once(':').with_whatever_context(|| {
+                            format!("{envvar} should be in 'username:password' format.")
+                        })?;
+                        match std::env::var("PASSWORDS_HASHED") {
+                            Ok(_) => (name, password.to_string()),
+                            _ => (
+                                name,
+                                // hashes password with fixed salt, because why bother properly
+                                // salting a password that is stored in plaintext anyways?
```

Comment:
I'd find a comment like the following more useful:

// Plain text passwords provided; hash them with a fixed salt.